docs/lambda_versioning.md

```diff
@@ -0,0 +1,71 @@
+# Lambda Versioning
+
+Cumulus makes use of AWS's Lambda/Alias version objects to tag and retain references to recent copies of deployed workflow lambdas.
+
+Each Cumulus deployed lambda and configured s3Source lambda in lambdas.yml will have an alias/version resource created for them.
+
+A reference to the most current lambda version alias will replace the unversioned lambda resource ARN in all workflows for each task that is either built via Cumulus, or defined via the uniqueIdentifier configuration key for s3 sourced lambdas.
+
+A configurable number of previously deployed alias/version pairs will be retained to ensure that in-progress workflows are able to complete.
+
+This allows for workflows to automatically reference the specific version of a lambda function they were deployed with, prevents an updated deployment of an existing lambda from being utilized in an already in-progress workflow, and retains the executed version information in the AWS step function execution record and cloud watch logs.
+
+**Please note** that care must be exercised to not update lambda versions and redeploy frequently enough that an in-progress workflow refers to an aged-off version of a lambda, or workflows that reference such a lambda may fail.
```

Comment:
What is frequently enough? Maybe we should say 2 versions are kept for each lambda so if you were to deploy 3x before a running workflow could complete, that's when you run into issues?

Reply:
This is tricky - because redeploying 3x isn't the issue, it's redeploying with an updated version of the lambda (deploying/rolling back should be fine infinitely).      

Frequently enough is "Deploy new versions quickly enough that you are > retained lambdas ahead of an executing workflow that has said lambda".    I'm extremely open to a better way to say that.

Reply:
Yeah my concern here is that it's going to scare people who read it haha. I was trying to figure out how not to make it scary since this would be super rare. I'm feeling like you can just leave it.